rest/data/patch.go

```diff
@@ -327,10 +331,20 @@ func (p *DBPatchConnector) GetPatchRawPatches(patchID string) (map[string]string
 // MockPatchConnector is a struct that implements the Patch related methods
 // from the Connector through interactions with he backing database.
 type MockPatchConnector struct {
-	CachedPatches    []restModel.APIPatch
-	CachedAborted    map[string]string
-	CachedPriority   map[string]int64
-	CachedRawPatches map[string]string
+	CachedPatches     []restModel.APIPatch
+	CachedProjectRefs []restModel.APIProjectRef
+	CachedAborted     map[string]string
+	CachedPriority    map[string]int64
+	CachedRawPatches  map[string]string
+}
+
+func (hp *MockPatchConnector) GetIdForProject(identifier string) (string, error) {
```

Comment:
I don't think you should make a separate connector function for this. The mock functions are meant to mock the function of the same name for the _real_ connector, and GetIdForProject is part of the FindPatchesByProject connector function, not its own function. I would embed this logic below.

Reply:
done